docs/advanced/server-side-rendering.md

```diff
@@ -246,6 +246,78 @@ When you are using `getStaticProps`, `getServerSideProps`, or `getStaticPaths`,
 During the prepass of your component tree `next-urql` can't know how these functions will alter the props passed to your page component. This injection
 could change the `variables` used in your `useQuery`. This will lead to error being thrown during the subsequent `toString` pass, which isn't supported in React 16.
 
+### Using getStaticProps or getServerSideProps
+
+By default `withUrqlClient` will add `getInitialProps` to the component you're wrapping it in, this however excludes us from using
+`getStaticProps` and `getServerSideProps`. However we can enable this, let's look at an example:
+
+```js
+import { withUrqlClient, initUrqlClient } from "next-urql";
+import {
+  ssrExchange,
+  dedupExchange,
+  cacheExchange,
+  fetchExchange,
+  useQuery
+} from "urql";
+
+const TODOS_QUERY = `
+  query { todos { id text } }
+`;
+
+const createUrqlClient = (ssr, ctx) => ({
+  url: "your-url"
+  exchanges: [dedupExchange, cacheExchange, ssrCache, fetchExchange]
+});
+
+function Todos() {
+  const [res] = useQuery({ query: TODOS_QUERY });
+  return (
+    <div>
+      {res.data.todos.map((todo) => (
+        <div key={todo.id}>
+          {todo.id} - {todo.text}
+        </div>
+      ))}
+    </div>
+  );
+}
+
+export async function getStaticProps(ctx) {
+  const ssrCache = ssrExchange({ isClient: false });
+  const client = initUrqlClient(createUrqlClient(ssrCache, ctx));
+
+  await client.query(TODOS_QUERY).toPromise();
```

Comment:
We should probably limit this piece of text by stating its drawbacks in regards of this being "useless" for caching, cache updates, etc

Reply:
nvm, actually, since this just mirrors the hook lol